### PR TITLE
AST-3387 - Fix: WooCommerce feature removal filter causing site crashing issue.

### PR DESCRIPTION
**Pull request by yssambare12**
**Description** - When we activate Astra that time it automatically adds an extra feature for WooCommerce in customization so if any user wants to remove this then we are providing a filter.

`add_filter( 'astra_enable_woocommerce_integration', '__return_false' );
`
https://wpastra.com/docs/remove-astra-customization-for-woocommerce/ 

**Issue** - Right now if we use this filter then its breaking the site.

**Steps To Reproduce:** 

1. Install and Activate the Astra + WooCommerce
2. Add the above filter and check the site 

**Screenshot** - https://prnt.sc/Gj-PpJqEIPbD

**Other comments:**

https://brainstormforce.slack.com/archives/CDS7TQ75Z/p1690882533372169

### Types of changes

- Code Improvement.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php [hunk 1]
```diff
@@ -497,10 +497,13 @@ function astra_hb_woo_cart_dynamic_css( $dynamic_css, $dynamic_css_filtered = ''
 	if ( 'none' !== $header_cart_icon_style ) {
 
 
-		if ( ! astra_has_pro_woocommerce_addon() && 'outline' === $header_cart_icon_style && 'default' !== $header_woo_cart_list ) {
-
+		if (
+			( ! function_exists( 'astra_has_pro_woocommerce_addon' ) || ! astra_has_pro_woocommerce_addon() ) &&
```
Comment (nathanrodrigues2111) [inline inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php:501]: Correct me if i am wrong but i think this condition should be like this 

if( function_exists( 'astra_has_pro_woocommerce_addon' &&  ! astra_has_pro_woocommerce_addon() && 'outline' === $header_cart_icon_style && 'default' !== $header_woo_cart_list )  {
}

Why is it ! function_exists ? . As we should be checking if the function exists  @yssambare12

**Review by nathanrodrigues2111 — COMMENTED** [review]
Please all the add changelogs if required

File: inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php (hunk 1 above)
Comment (yssambare12) [reply inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php:501]: Yes, @nathanrodrigues2111 you are right! Let me make these changes.

Comment (yssambare12) [reply inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php]: Updated!

**Comment by yssambare12** [comment]
> Please all the add changelogs if required

I guess a changelog is not necessary here - The issue was noticed while using the [Doc](https://wpastra.com/docs/remove-astra-customization-for-woocommerce/) mention filter.

**Review by nathanrodrigues2111 — APPROVED** [review]

File: inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php [hunk 2]
```diff
@@ -497,10 +497,12 @@ function astra_hb_woo_cart_dynamic_css( $dynamic_css, $dynamic_css_filtered = ''
 	if ( 'none' !== $header_cart_icon_style ) {
 
 
-		if ( ! astra_has_pro_woocommerce_addon() && 'outline' === $header_cart_icon_style && 'default' !== $header_woo_cart_list ) {
-
+		if ( function_exists( 'astra_has_pro_woocommerce_addon' ) && ! astra_has_pro_woocommerce_addon() &&
+			'outline' === $header_cart_icon_style &&
+			'default' !== $header_woo_cart_list
+		) {
```
Comment (imnavanath) [inline inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php:503]:
```suggestion
		if ( function_exists( 'astra_has_pro_woocommerce_addon' ) && ! astra_has_pro_woocommerce_addon() && 'outline' === $header_cart_icon_style && 'default' !== $header_woo_cart_list ) {
```

File: inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php [hunk 3]
```diff
@@ -497,10 +497,9 @@ function astra_hb_woo_cart_dynamic_css( $dynamic_css, $dynamic_css_filtered = ''
 	if ( 'none' !== $header_cart_icon_style ) {
 
 
-		if ( ! astra_has_pro_woocommerce_addon() && 'outline' === $header_cart_icon_style && 'default' !== $header_woo_cart_list ) {
-
+		if ( function_exists( 'astra_has_pro_woocommerce_addon' ) && ! astra_has_pro_woocommerce_addon() && 'outline' === $header_cart_icon_style && 'default' !== $header_woo_cart_list ) {
 			$border_width = astra_get_option( 'woo-header-cart-border-width' );
-
+
```
Comment (imnavanath) [inline inc/builder/type/header/woo-cart/dynamic-css/dynamic.css.php:502]:
```suggestion

```

File: inc/core/class-astra-enqueue-scripts.php [hunk 4]
```diff
@@ -385,14 +385,15 @@ public function enqueue_scripts() {
...
+			
 				$astra_shop_add_to_cart_localize_data = array(
 					'shop_add_to_cart_action' => astra_get_option( 'shop-add-to-cart-action' ),
 					'cart_url'                => wc_get_cart_url(),
 					'checkout_url'            => wc_get_checkout_url(),
-					'is_astra_pro'            => astra_has_pro_woocommerce_addon(),
+					'is_astra_pro'            => $is_astra_pro,
 				);
-
+
```
Comment (imnavanath) [inline inc/core/class-astra-enqueue-scripts.php:396]:
```suggestion

```

**Review by imnavanath — APPROVED** [review]

**Comment by imnavanath** [comment]
@yssambare12 Please add changelog for this.

**Comment by yssambare12** [comment]
> @yssambare12 Please add changelog for this.

Changelog Added.

File: changelog.txt [hunk 5]
```diff
@@ -24,6 +24,7 @@ v4.1.7 (Unreleased)
 - Fix: WooCommerce - Elementor Pro's Mini Cart widget conflicts with Astra Mini Cart builder element.
 - Fix: WooCommerce - Left margin from the "Add to cart" button when Sold Individually option is selected.
 - Fix: WooCommerce - Shop - Slide in cart action for Add to Cart button does not work when Elementor is active.
+- Fix: WooCommerce - Site crashing issue when using WooCommerce extended feature removal filter.
```
Comment (imnavanath) [inline changelog.txt:27]:
```suggestion
- Fix: WooCommerce - PHP fatal error when 'astra_enable_woocommerce_integration' filter used to disable Astra and WooCommerce extended integration. ( https://wpastra.com/docs/remove-astra-customization-for-woocommerce/ )
```

**Pull request merged**